### PR TITLE
Remove Unused Settings - Part 1

### PR DESCRIPTION
**Pull request by stevenjcumming**
## Summary

This is part 1 of 3-ish

- Remove settings from `config/settings.yml` and `config/settings/test.yml`
- Checked using:
   - Dot notation
   - Hash access (`[]`, `.dig`, `fetch`)
   - Checked parent key
- Added comments for difficult to find Settings (willing to remove if undesirable)

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/95565

## Testing done

- [ ] n/a

## Acceptance criteria

- [x]  Unused settings are removed from `config/settings` files

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -86,12 +86,10 @@ sign_in:
 
 terms_of_use:
   current_version: v1
-  provisioner_cookie_domain: localhost
```
Comment (rmtolmach) [inline config/settings.yml:89]: Riley added this last year in https://github.com/department-of-veterans-affairs/vets-api/issues/14455. @rileyanderson, we don't see `Settings.terms_of_use.provisioner_cookie_domain` called anywhere. I just wanted to check with you before we deleted this.

**Comment by rmtolmach** [comment]
Seems good to me. I asked for Riley's review because that key is relatively new compared to the others you're removing.  Are any of these values in the manifest repo?

**Comment by stevenjcumming** [comment]
@mathisto could you take a look at this PR regarding the updates to appeals_api specs (or anything else if you see something)

File: config/settings.yml (hunk 1 above)
Comment (stevenjcumming) [reply config/settings.yml:89]: Talked to Riley and `provisioner_cookie_domain` isn't being used

File: config/settings.yml [hunk 2]
```diff
@@ -387,8 +378,6 @@ modules_appeals_api:
       enabled: false
     weekly_decision_review:
       enabled: false
-    daily_error:
-      enabled: false
     weekly_error:
```
Comment (mathisto) [inline config/settings.yml:381]: I believe you can drop this key and it child for the same reasons you dropped the daily.

File: modules/appeals_api/spec/sidekiq/higher_level_review_upload_status_batch_spec.rb [hunk 3]
```diff
@@ -35,11 +35,9 @@
         let(:upload) { create(:higher_level_review_v1, status: 'received') }
 
         it 'ignores them' do
-          with_settings(Settings.modules_appeals_api, higher_level_review_updater_enabled: true) do
```
Comment (mathisto) [inline modules/appeals_api/spec/sidekiq/higher_level_review_upload_status_batch_spec.rb:38]: I believe that unlike the subsequent files, this batch version's `Settings` will need to be replaced with the relevant Flipper flag.

**Review by mathisto — CHANGES_REQUESTED** [review]

File: modules/appeals_api/spec/sidekiq/higher_level_review_upload_status_batch_spec.rb (hunk 3 above)
Comment (stevenjcumming) [reply modules/appeals_api/spec/sidekiq/higher_level_review_upload_status_batch_spec.rb:38]: @mathisto are you saying you would like me to leave `with_settings...` in the spec?

Comment (rmtolmach) [reply modules/appeals_api/spec/sidekiq/higher_level_review_upload_status_batch_spec.rb:38]: @mathisto Can you be more specific about what changes you want here?

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**